noir-projects/aztec-nr/aztec/src/utils/bytes.nr

```diff
@@ -1,88 +1,319 @@
+            chunk[j] = random_bits[i * NUM_BITS + j];
+        }
+        chunks[i] = chunk;
+    }
+    chunks
+}
+
+/**
+ * Helper fn for be/le_bytes_to_padded_fields() fns.
+ */
+fn pad_31_byte_fields_with_random_bits<let N: u32>(input: [Field; N]) -> [Field; N] {
+    /// Safety: we can't constrain whether or not the returned bits are actually random;
+    /// we can only rely on it being in the caller's best interest.
+    let chunks_of_random_bits = unsafe { get_chunks_of_random_bits::<((N + 30) / 31), 6>() };
+    let mut fields = input;
+    // We push the random bits to the "big end" of each field
+    let TWO_POW_248 = 0x100000000000000000000000000000000000000000000000000000000000000;
+    for i in 0..fields.len() {
+        let mut shift = TWO_POW_248;
+        for j in 0..5 {
+            fields[i] += shift * chunks_of_random_bits[i][j] as Field;
+            shift *= 2;
+        }
+
+        // This might overflow the field modulus, so we need to be careful.
+        // By this point, shift = 2 ^ 253, so will populate the 254th bit (counting from 1)
+        let field_with_254th_bit = fields[i] + shift * chunks_of_random_bits[i][6 - 1] as Field;
+
+        // Expensive comparison (one for each field), but it's important -- if we want to hide that we were
+        // working with bytes -- that we include the 254th random bit when it fits.
+        fields[i] = if (fields[i].lt(field_with_254th_bit)) {
+            field_with_254th_bit
+        } else {
+            fields[i]
+        };
+    }
+    fields
+}
+
+/**
+ * The above functions be_bytes_31_to_fields and le_bytes_31_to_fields have a privacy-leakage problem:
+ * they result in fields which are all 248-bits instead of 254-bits.
+ * That means if the above functions are used to convert a ciphertext from bytes to
+ * fields, it leaks (with very high probability) to all observers of the log that the
+ * ciphertext was created with a "bit twiddly" encryption scheme like AES.
+ * This reduces privacy sets for all apps.
+ *
+ * This function seeks to "pad" those remaining 6 bits with randomness, so that all
+ * encrypted logs look even more similar (full fields of 254-bits of random-looking
+ * information). Of course, this adds some extra constraints, unfortunately.
+ */
+pub fn le_bytes_to_padded_fields<let N: u32>(input: [u8; N]) -> [Field; (N + 30) / 31] {
+    let mut fields = le_bytes_31_to_fields(input);
+    pad_31_byte_fields_with_random_bits(fields)
+}
+
+pub fn be_bytes_to_padded_fields<let N: u32>(input: [u8; N]) -> [Field; (N + 30) / 31] {
+    let mut fields = be_bytes_31_to_fields(input);
+    pad_31_byte_fields_with_random_bits(fields)
 }
 
 mod test {
-    use crate::utils::bytes::{bytes_to_fields, fields_to_bytes};
+    use crate::utils::bytes::{
+        be_bytes_31_to_fields, fields_to_be_bytes_31, fields_to_le_bytes_31, le_bytes_31_to_fields,
+    };
+
+    // BE BYTES
 
     #[test]
-    fn test_bytes_to_1_field() {
+    fn test_be_bytes_31_to_1_field() {
```

Comment:
You can remove the `test_` preffix and this will show up as `bytes::test::be_bytes_31_...` in the test summary anyway because the module is called `test`.

Reply:
I figured it was some kind of unwritten convension, so I followed suit, but sure.